### PR TITLE
[coqdep] moved coqdeplib into own directory

### PR DESCRIPTION
**Pull request by Alizter**
We do some refactoring on coqdep. Notably, we separate the parsing and initialisation phases of coqdep, clearing the ambiguity with coqdeplib that was there before.

This will allow for further cleanup and refactoring.

cc @ejgallego

### CODE REVIEW

**Review by ejgallego — COMMENTED** [review]
Looks good, maybe the directory could just be `coqdep/lib` ?

Having a top-level error and args module is fine, however not very useful to define interfaces for them if in the end they are called from only one module.

In general, I think it makes sense to split a module to its own file when it has more than one user, large surface API has a large cost IMHO.

**Comment by Alizter** [comment]
@ejgallego I've renamed coqdeplib to lib and removed the toplevel error interface. However the args module is important since this is what you pass to initialize coqdep.

We can do some more cleanups of what is in the API soon when we begin using it ourselves.

File: tools/coqdep/lib/args.ml [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+(*  v      *         Copyright INRIA, CNRS and contributors             *)
+(* <O___,, * (see version control and CREDITS file for authors & dates) *)
+(*   \VV/  **************************************************************)
+(*    //   *    This file is distributed under the terms of the         *)
+(*         *     GNU Lesser General Public License Version 2.1          *)
+(*         *     (see LICENSE file for the text of the license)         *)
+(************************************************************************)
+
+type t =
+{ boot : bool
```
Comment (ejgallego) [inline tools/coqdep/lib/args.ml:12]: Is the indentation here ok?

File: tools/coqdep/lib/options.mli [hunk 2]
```diff
@@ -8,4 +8,12 @@
...
 
+module Dynlink : sig
+  type t = Opt | Byte | Both | No | Variable
+end
+
 val boot : bool ref
+val sort : bool ref
+val write_vos : bool ref
+val noglob : bool ref
+val dynlink : Dynlink.t ref
```
Comment (ejgallego) [inline tools/coqdep/lib/options.mli:19]: At some point in a future PR we could remove these and have the clients use `Args.t` directly.

**Review by ejgallego — APPROVED** [review]

File: tools/coqdep/lib/args.ml (hunk 1 above)
Comment (Alizter) [reply tools/coqdep/lib/args.ml]: Missed that thanks.

File: tools/coqdep/lib/options.mli (hunk 2 above)
Comment (Alizter) [reply tools/coqdep/lib/options.mli:19]: @ejgallego They can initialize Coqdep.Lib with Args.t directly, thats what init does now. These options are simply the internal state of coqdep. We could probably move this to a common state file.

**Comment by ejgallego** [comment]
Will merge soon, unless you want to squash some history.

**Comment by Alizter** [comment]
I won't squash this since it is tricky to undo these commits otherwise.

**Comment by ejgallego** [comment]
Yeah no need to squash, however for example the renaming and the indentation fix could disappear easily using `rebase -i` and make a bit better history.

**Comment by Alizter** [comment]
@ejgallego ping

**Review by ejgallego — APPROVED** [review]
Thanks

**Comment by ejgallego** [comment]
@coqbot: merge now

**Pull request merged**